cs/src/Connections/ITunnelClient.cs

```diff
@@ -94,5 +94,18 @@ Task ConnectAsync(Tunnel tunnel, CancellationToken cancellation)
         /// <returns>A <see cref="Task{Stream}"/> representing the result of the asynchronous operation.</returns>
         /// <exception cref="InvalidOperationException">If the tunnel is not yet connected and hasn't started connecting.</exception>
         Task<Stream?> ConnectToForwardedPortAsync(int forwardedPort, CancellationToken cancellation);
+
+        /// <summary>
+        /// Event handler for refreshing the tunnel access token.
+        /// The tunnel client will fire this event when it is not able to use the access token it got from the tunnel.
+        /// </summary>
+        event EventHandler<RefreshingTunnelAccessTokenEventArgs>? RefreshingTunnelAccessToken;
+
+        /// <summary>
+        /// SSH session reconnected event.
+        /// If the session doesn't support reconnects, the client may try to start a brand new SSH session.
+        /// This event is fired after either reconnection succeeds or a new session is created.
+        /// </summary>
+        event EventHandler SshSessionReconnected;
```

Comment:
Should there be a `Disconnected` event? Or combine them as `ConnectionStatusChanged`?

Also it might be better to drop the `SshSession` prefix, to hide the SSH implementation details as much as possible, in case we ever want to support a new non-SSH connection protocol.

Reply:
I see there is `TunnelClientBase.SshSessionClosed`; maybe that can be merged or at least made consistent.